### ISSUE TITLE
[JENKINS-33930] new tabbed job config page broken accessibility

### ISSUE THREAD

**Issue by jenkins-infra-bot** [timja-org/jenkins-gh-issues-poc-11-23#5630]
<p>you can no longer use the keyboard to navigate on the restyled "FreeStyle" job configuration page.</p>

<p>There is no way to change the tabs which would confuse a user (they are not tabs - but look like tabs? - when an item gets focus via a tab key the "tab" thing should change highlight as it does if you just scroll<br/>
The tabbing order of the items and help is completely out of whack.<br/>
The red X on a build step never seems to receive focus</p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/teilo.png" title="teilo's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=teilo">teilo</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-33930" target="_blank">new tabbed job config page broken accessibility</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/recena.png" title="recena's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=recena">recena</a>
<li><b>status</b>: Resolved
<li><b>priority</b>: Critical
<li><b>component(s)</b>: core
<li><b>label(s)</b>: 2.0, 2.0-beta, accessibility, community-bee, ui
<li><b>resolution</b>: Cannot Reproduce
<li><b>resolved</b>: 2018-01-24T22:08:07+00:00
<li><b>votes</b>: 1
<li><b>watchers</b>: 8
<li><b>imported</b>: 2025-11-23
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>you can no longer use the keyboard to navigate on the restyled "FreeStyle" job configuration page.</p>

<p>There is no way to change the tabs which would confuse a user (they are not tabs - but look like tabs? - when an item gets focus via a tab key the "tab" thing should change highlight as it does if you just scroll
The tabbing order of the items and help is completely out of whack.
The red X on a build step never seems to receive focus</p></pre>
</details>
</details>
<details><summary><i>1 attachment</i></summary>

- [Screenshot 2016-05-30 10.59.05.png](https://issues.jenkins.io/secure/attachment/32827/Screenshot%202016-05-30%2010.59.05.png)
  > ![Screenshot 2016-05-30 10.59.05.png](https://issues.jenkins.io/secure/attachment/32827/Screenshot%202016-05-30%2010.59.05.png)
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-33930?focusedId=252614&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-252614">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<blockquote><p>The red X on a build step never seems to receive focus,</p></blockquote>

<p>It's still the last element (even though visually it's first), but could probably use a focus highlight.</p>

<blockquote><p>The tabbing order of the items and help is completely out of whack.</p></blockquote>

<p>Do you mean here that tab focus jumps to help icons when adding an item to a hetero-list. If not, I don't understand what the problem is.</p>
</pre>
</details>

</ul>
</details>

<blockquote><p>The red X on a build step never seems to receive focus,</p></blockquote>

<p>It's still the last element (even though visually it's first), but could probably use a focus highlight.</p>

<blockquote><p>The tabbing order of the items and help is completely out of whack.</p></blockquote>

<p>Do you mean here that tab focus jumps to help icons when adding an item to a hetero-list. If not, I don't understand what the problem is.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/teilo.png" title="teilo's avatar" /> teilo:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-33930?focusedId=253008&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-253008">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<blockquote><p>If the RedX is not visually highlighted in 2.0 how do you know it is selected - how do you know you have navigated to it? </p></blockquote>
</pre>
</details>

</ul>
</details>

<blockquote><p>If the RedX is not visually highlighted in 2.0 how do you know it is selected - how do you know you have navigated to it? </p></blockquote>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>gusreiber:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-33930?focusedId=253505&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-253505">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>....waiting for gitHub.</p>
</pre>
</details>

</ul>
</details>

<p>....waiting for gitHub.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-33930?focusedId=253507&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-253507">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in jenkins
User: gusreiber
Path:
 war/src/main/js/widgets/config/tabbar.less
<a href="http://jenkins-ci.org/commit/jenkins/99e482b2a94102c6c2b3445c2fa1514f242fede5" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/99e482b2a94102c6c2b3445c2fa1514f242fede5</a>
Log:
  <a href="https://issues.jenkins.io/browse/JENKINS-33930" title="new tabbed job config page broken accessibility" class="issue-link" data-issue-key="JENKINS-33930"><del>JENKINS-33930</del></a> focus color for delete</p>

<p>style only.</p>


</pre>
</details>

</ul>
</details>

<p>Code changed in jenkins<br/>
User: gusreiber<br/>
Path:<br/>
 war/src/main/js/widgets/config/tabbar.less<br/>
<a href="http://jenkins-ci.org/commit/jenkins/99e482b2a94102c6c2b3445c2fa1514f242fede5" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/99e482b2a94102c6c2b3445c2fa1514f242fede5</a><br/>
Log:<br/>
  <a href="https://issues.jenkins.io/browse/JENKINS-33930" title="new tabbed job config page broken accessibility" class="issue-link" data-issue-key="JENKINS-33930"><del>JENKINS-33930</del></a> focus color for delete</p>

<p>style only.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/aheritier.png" title="aheritier's avatar" /> aheritier:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-33930?focusedId=259303&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-259303">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Is it a known issue the rendering of the delete button in config panels ?
<span class="image-wrap" style=""><a id="32827_thumb" href="https://issues.jenkins.io/secure/attachment/32827/32827_Screenshot+2016-05-30+10.59.05.png" title="Screenshot 2016-05-30 10.59.05.png" file-preview-type="image" file-preview-id="32827" file-preview-title="Screenshot 2016-05-30 10.59.05.png"><img src="https://issues.jenkins.io/secure/thumbnail/32827/_thumb_32827.png" style="border: 0px solid black" role="presentation"/></a></span>
I didn't find it reported but I saw it for a long time</p>
</pre>
</details>

</ul>
</details>

<p>Is it a known issue the rendering of the delete button in config panels ?<br/>
<span class="image-wrap" style=""><a id="32827_thumb" href="https://issues.jenkins.io/secure/attachment/32827/32827_Screenshot+2016-05-30+10.59.05.png" title="Screenshot 2016-05-30 10.59.05.png" file-preview-type="image" file-preview-id="32827" file-preview-title="Screenshot 2016-05-30 10.59.05.png"><img src="https://issues.jenkins.io/secure/thumbnail/32827/_thumb_32827.png" style="border: 0px solid black" role="presentation"/></a></span><br/>
I didn't find it reported but I saw it for a long time</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/aheritier.png" title="aheritier's avatar" /> aheritier:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-33930?focusedId=259315&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-259315">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Ok I just found <a href="https://issues.jenkins.io/browse/JENKINS-35178" title="&quot;Delete build step&quot; button has bad layout in Safari" class="issue-link" data-issue-key="JENKINS-35178"><del>JENKINS-35178</del></a> and PR <a href="https://github.com/jenkinsci/jenkins/pull/2378" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/pull/2378</a> by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=recena" class="user-hover" rel="recena">recena</a></p>
</pre>
</details>

</ul>
</details>

<p>Ok I just found <a href="https://issues.jenkins.io/browse/JENKINS-35178" title=""Delete build step" button has bad layout in Safari" class="issue-link" data-issue-key="JENKINS-35178"><del>JENKINS-35178</del></a> and PR <a href="https://github.com/jenkinsci/jenkins/pull/2378" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/pull/2378</a> by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=recena" class="user-hover" rel="recena">recena</a></p>
